admin-dev/themes/new-theme/js/pages/product/combination/bulk-form-handler.ts

```diff
@@ -0,0 +1,170 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ConfirmModal from '@components/modal';
+import ProductMap from '@pages/product/product-map';
+import ComponentsMap from '@components/components-map';
+import ProductEvents from '@pages/product/product-event-map';
+import CombinationsService from '@pages/product/services/combinations-service';
+import {EventEmitter} from 'events';
+import DisablingToggler from '@components/form/disabling-toggler';
+
+const CombinationMap = ProductMap.combinations;
+const CombinationEvents = ProductEvents.combinations;
+
+export default class BulkFormHandler {
+  private combinationsService: CombinationsService;
+
+  private eventEmitter: EventEmitter;
+
+  private tabContainer: HTMLDivElement;
+
+  constructor() {
+    this.combinationsService = new CombinationsService();
+    this.eventEmitter = window.prestashop.instance.eventEmitter;
+    this.tabContainer = document.querySelector(CombinationMap.externalCombinationTab) as HTMLDivElement;
+    this.init();
+  }
+
+  private init(): void {
+    this.listenSelections();
+    this.eventEmitter.on(CombinationEvents.listRendered, () => this.toggleBulkActions());
+
+    const bulkFormBtn = document.querySelector(CombinationMap.bulkCombinationFormBtn) as HTMLButtonElement;
+    bulkFormBtn.addEventListener('click', () => this.showFormModal());
+  }
+
+  private showProgressModal(): ConfirmModal {
+    //@todo: Replace with new progress modal when introduced in #26004.
+    const modal = new ConfirmModal(
+      {
+        id: CombinationMap.bulkProgressModalId,
+        confirmMessage: '<div>Updating combinations: <p class="progress-increment"></p></div>',
+      },
+      () => null,
+    );
+
+    modal.show();
+
+    return modal;
+  }
+
+  private showFormModal(): ConfirmModal {
+    const template = document.querySelector(CombinationMap.bulkCombinationFormTemplate) as HTMLScriptElement;
+    const content = template.innerHTML;
+
+    const modal = new ConfirmModal(
+      {
+        id: CombinationMap.bulkCombinationModalId,
+        confirmMessage: content,
+      },
+      () => this.submitForm(),
+    );
+
+    modal.show();
+    const form = document.querySelector(CombinationMap.bulkCombinationForm) as HTMLFormElement;
+    const disablingToggles = form.querySelectorAll(ComponentsMap.disablingToggle.wrapper) as NodeListOf<HTMLDivElement>;
+
+    disablingToggles.forEach((element) => {
+      const {disablingToggleName} = element.dataset;
+      new DisablingToggler(
+        `${CombinationMap.bulkCombinationForm} [data-disabling-toggle-name="${disablingToggleName}"] input`,
```

Comment:
I'm not sure you need the `data-toggled-by` attribute since you are already looping through the toggling inputs you can use `element.id` here

This will simplify this code and the listener code as well since you won't need to add this extra data attribute anymore The `data-disabling-toggle-name` data attribute is necessary however But the selector for disabled input can be simplified as well (see next comment)